source/common/router/upstream_request.cc

```diff
@@ -235,9 +235,11 @@ class UpstreamFilterManager : public Http::FilterManager {
 };
 
 UpstreamRequest::UpstreamRequest(RouterFilterInterface& parent,
+                                 const Http::RequestHeaderMap& request_headers,
                                  std::unique_ptr<GenericConnPool>&& conn_pool,
                                  bool can_send_early_data, bool can_use_http3)
-    : parent_(parent), conn_pool_(std::move(conn_pool)), grpc_rq_success_deferred_(false),
+    : parent_(parent), request_headers_(request_headers), conn_pool_(std::move(conn_pool)),
+      grpc_rq_success_deferred_(false),
```

Comment:
This change also unnecessary. The upstream request can access request headers by `downstreamHeaders()` of `RouterFilterInterface`.

Reply:
Reverted to the  original form.